accord-core/src/main/java/accord/messages/ReadData.java

```diff
@@ -359,11 +359,13 @@ void cancel(@Nullable SafeCommandStore safeStore)
             if (safeCommand != null) safeCommand.removeListener(this);
             waitingOn.clear(safeStore.commandStore().id());
         }
+
         // TODO (expected): efficient unsubscribe mechanism
-        node.commandStores().mapReduceConsume(this, waitingOn.stream(), forEach(in -> {
-            SafeCommand safeCommand = in.ifInitialised(txnId);
-            if (safeCommand != null) safeCommand.removeListener(this);
-        }, node.agent()));
+        if (waitingOn != null)
```

Comment:
did you hit a NPE?  since we touch `waitingOn.stream()` we would hit a NPE.

Reply:
yes, if there are multiple calls to cancel it will throw an NPE